pkg/conf/schedulerconf.go

```diff
@@ -45,7 +47,7 @@ const (
 	DefaultKubeBurst            = 1000
 )
 
-var configuration *SchedulerConf
+var pointer unsafe.Pointer
```

Comment:
Could you give rationale why the `unsafe` should be used here?

Reply:
I do not see any reason why we should use unsafe. Unsafe has its uses but I think  this is not one of them.
Unsafe is not stable and can change in any go release.

The other problem is with the `Set`.  `Init` will always be called before we set a new one. If we hold on to the previous pointer that we got via a `GetSchedulerConf()` then that could cause issues.
Atomic also does not provide any ordering guarantee, it just provides the correct value at that point in time. 

`Set` is only called in tests. Instead of replacing pointer we should update the content of the structure created in `init`. That means the pointer will never change. I would rather go to a full get/set structure for the config.

Reply:
Thanks, @adamantal @wilfred-s 
this a fair point, I removed unsafe pkg used in this patch. Instead, using a lock and get/set approach.
Since the conf object is global, I think we should not replace the object anytime, just stick to update or get values.
Please see my latest commit. Thanks!